### PR TITLE
Fixed GTG and Health check.

### PR DESCRIPTION
**Pull request by carlosroman**
Also changed the behaviour so that GTG is based on status of DB.

### CODE REVIEW

File: people/service.go [hunk 1]
```diff
@@ -56,15 +58,45 @@ func (s *peopleServiceImpl) isInitialised() bool {
...
+}
+
 func (s *peopleServiceImpl) Shutdown() error {
-	log.Info("Shutingdown...")
+	log.Info("Shuting down...")
+	s.Lock()
+	s.initialised = false
+	s.dataLoaded = false
+	defer func() {
+		s.Unlock()
```
Comment (peteclark-ft) [inline people/service.go]: Can you not do `defer s.Unlock()` ?

File: people/service.go [hunk 2]
```diff
@@ -100,46 +136,49 @@ func (s *peopleServiceImpl) getPersonByUUID(uuid string) (person, bool, error) {
...
-		log.Errorf("ERROR unmarshalling cached value for [%v]: %v", uuid, err.Error())
+		log.Errorf("ERROR unmarshalling cached value for [%v]: %v.", uuid, err.Error())
 		return person{}, true, err
 	}
 	return cachedPerson, true, nil
 
 }
 
 func (s *peopleServiceImpl) openDB() error {
+	s.Lock()
```
Comment (peteclark-ft) [inline people/service.go:151]: I don't like that this is expected to be unlocked by `createCacheBucket` if the bucket isn't there... Having said that, I assume you want to keep the lock for the whole interaction - maybe add a comment?

File: people/service.go [hunk 3]
```diff
@@ -194,20 +233,27 @@ func (s *peopleServiceImpl) processPeople(c <-chan []person, wg *sync.WaitGroup)
...
-	log.Info("Finished processing all people")
-	wg.Done()
+	log.Info("Finished processing all people.")
+	if s.isInitialised() {
+		s.setDataLoaded(true)
+	}
 }
 
 func (s *peopleServiceImpl) createCacheBucket() error {
+	defer s.Unlock()
```
Comment (peteclark-ft) [inline people/service.go]: Same here - worth commenting that this has to happen.

**Review by peteclark-ft — APPROVED** [review]
👍

File: people/service.go (hunk 1 above)
Comment (carlosroman) [reply people/service.go]: yeah, originally I had a lot of things happening in there but on last refactor I didn't refactor away the entire function... will update :)

File: people/service.go (hunk 2 above)
Comment (carlosroman) [reply people/service.go:151]: The defer is now in this function so more obvious about what unlocks it. Tested it without the lock but figured no harm in keeping it there.

**Pull request merged**